### ISSUE TITLE
[Bug]: dropbox folder name is case preserving by creation

### ISSUE THREAD

**Issue by daledesilva** [remotely-save/remotely-save#567]
### What happened?

After updating to version 4.11 on both devices, I'm now receiving these notifications:
```
remotely-save-1711680494617: abort sync, triggerSource=manual, error while generating_plan
-
unexpectedly keptFolder no decisions: ink/
```
I'm on Mac M1 and iPad.

I've tried renaming the folder on my desktop but it doesn't help.
I've also deleted Sync Plans History, deleted Prev Sync Details, and Reset Cache/Databases but no luck.

I've alsp deleted the vaults folder on dropbox apps/remotely-save/
This stopped my desktop aborting due to error but my iPad still does, even after deleting plans cache, and reinstall plugin again.

I previously posted [this issue](https://github.com/remotely-save/remotely-save/issues/406) which seems related.

I really appreciate your work in building this and updating it recently.
Unfortunately, if a single error can cause the whole sync to abort entirely,  then for those who experience that error, the plugin becomes unusable.

Hopefully you can at least prevent the abort soon, or hopefully I can get some time to do a pull request too.

### What OS are you using?

macOS, iOS (iPhone, iPad)

### What remote cloud services are you using? (Please choose the specified one if it's in the list)

Dropbox

### Version of the plugin

4.11

### Version of Obsidian

1.5.11

### Using password or not

- [ ] Yes.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by fyears** [comment]
may i ask what's your folder structure in local and dropbox? especially what's "around" the folder `ink/`?

**Comment by daledesilva** [comment]
This is what dropbox shows in the remotely-saved managed folder:
<img width="324" alt="image" src="https://github.com/remotely-save/remotely-save/assets/18710973/f437586a-9513-4ab0-bdcd-818b8e6f4883">

However I also previously renamed the ink folder to something else and deleted vault folder entirely and resync'd, both times I received the same issue.

I've since tried deleting the vault in obsidian on my iPad, closing Obsidian and opening it again, and then recreating it with the same name but upon installing remotely-save again and syncing, I got the same error. Tried clearing cache and plans again, but same thing.
Note, the error is about 'ink/' lowercase, not 'Ink/' uppercase (which is what exists in dropbox).
At some point in the past the lowercase has existed and now doesn't, anywhere.

Lastly, I've now gotten around this by renaming by vault on desktop, then creating a new vault of the new name on iPad. This works again because it's completely new to Remotely-save.

I suspect that even when clearing caches, and uninstalling, there is some local memory on the iPad that isn't getting cleared.

**Comment by daledesilva** [comment]
I've just gone to the older vault on my iPad and tried changing the remotely save direction to 'incremental pull'... but still get the same error

**Comment by fyears** [comment]
do you have ink or Ink currently in your vault?

**Comment by fyears** [comment]
have you tried the "delete prev sync details in db" **in the new version**? i have improved this and hope i can clear the data as expected.

**Comment by daledesilva** [comment]
In some of those tests above I had already wiped my whole local vault (it was empty). But the error still persisted.

Yep, I'm in the new version and tried deleting details and history and clearing cache. But no luck.

Let me know if there's any log I can send you that will help.

**Comment by fyears** [comment]
yes! please sync and go to settings (near the end), then click "export sync plans", then delete any sensitive information and only keep all the entries about ink or sub files of ink and post them here. thank you very much!!

here is the old doc but the steps should be the same: https://github.com/remotely-save/remotely-save/blob/master/docs/how_to_debug/export_sync_plans.md

**Comment by fyears** [comment]
Never mind, no need to upload the sync plan for this issue.

I've done some experiment on dropbox and found out the root cause.

1. on dropbox website, create folder with upper case `Ink/` and file in the folder `Ink/a.md`.
2. on dropbox website, rename the folder to lower case `ink/`  (keeping the sub file `a.md` untouched.
3. in remotely-save, my code call the api of dropbox, which should return the file lists to me:

    ```json
    remoteEntityList:
    [
        {
            "keyRaw": "ink/",
            ...
        },
        {
            "keyRaw": "Ink/1.md",  // <- wait, what??
            ...
        }
    ]
    ```

4. As you might see, here comes the **inconsistency** between the **folder name** and the **file path**!!!

5. Magically, **the inconsistency preserves** even you rename `ink/` -> `some_name_else/` -> `ink/`, the **file path keeps being upper case `Ink/1.md`. I even delete the folder and recreate the folder on dropbox website and the api returns the same result..

6. I believe that's because dropbox is **case insensitive** but **case preserving**, and **case preserving under some logics**...

7. That's really annoying and I am thinking of a solution or workaround. I don't expect dropbox and fix this (you know, big corps...). I have to add some case detection.

**Comment by fyears** [comment]
Related: https://www.dropboxforum.com/t5/Dropbox-API-Support-Feedback/Case-Sensitivity-in-API-2/td-p/191279

So yeah  "the last path component (only) is guaranteed to have the expected casing". I have to build some logic to fix the path by components...

**Comment by daledesilva** [comment]
You're amazing!

**Comment by fyears** [comment]
merged and will be released in 0.4.12 (in a few days)
